ledger/participant-integration-api/src/main/scala/platform/store/FlywayMigrationsBase.scala

```diff
@@ -0,0 +1,294 @@
+  //   5. Update the value below
+  protected def baselineVersion: String
+
+  protected def getDataSource: Future[DataSource]
+
+  protected def locations: List[String]
+
+  private def runF[T](t: FluentConfiguration => Future[T]): Future[T] =
+    getDataSource.flatMap(dataSource => t(configurationBase(dataSource)))
+
+  private def run[T](t: FluentConfiguration => T): Future[T] =
+    runF(fc => Future(t(fc)))
+
+  private def configurationBase(dataSource: DataSource): FluentConfiguration =
+    Flyway
+      .configure()
+      .locations(locations: _*)
+      .dataSource(dataSource)
+
+  private def getStatus(flyway: Flyway): FlywayStatus = {
+    val info = flyway.info()
+    val current = info.current()
+    val pending = info.pending()
+    val applied = info.applied()
+
+    def isBeforeBaseline(m: MigrationInfo) = !m.getVersion.isAtLeast(baselineVersion)
+
+    // The Flyway history is immutable.
+    // To delete a migration, a new row is inserted into the history table that marks the migration as deleted.
+    // We are looking for rows that have not been marked as deleted by other rows (not DELETED),
+    // and that are not themselves deletion markers (not DELETE).
+    val notDeletedMigrationsBeforeBaseline = applied.toList
+      .filter(m =>
+        isBeforeBaseline(
+          m
+        ) && m.getType != MigrationType.DELETE && m.getState != MigrationState.DELETED
+      )
+      .map(_.getVersion.getVersion)
+
+    if (current == null) {
+      FlywayStatus.DatabaseEmpty(info.pending().length)
+    } else if (isBeforeBaseline(current)) {
+      FlywayStatus.SchemaBeforeBaseline(current.getVersion.getVersion, baselineVersion)
+    } else if (notDeletedMigrationsBeforeBaseline.nonEmpty) {
+      FlywayStatus.NeedsBaselineReset(notDeletedMigrationsBeforeBaseline)
+    } else if (pending.nonEmpty) {
+      FlywayStatus.Outdated(current.getVersion.getVersion, applied.length, pending.length)
+    } else {
+      FlywayStatus.NoMigrationsPending(current.getVersion.getVersion, applied.length)
+    }
+  }
+
+  private def resetBaseline(flyway: Flyway): Unit = {
+    val info = flyway.info()
+
+    assert(
+      info.current().getVersion.isNewerThan(baselineVersion),
+      "Current version is below the baseline, can't reset the baseline",
+    )
+    assert(
+      info.applied().exists(m => !m.getVersion.isNewerThan(baselineVersion)),
+      "No migrations below baseline found, baseline reset not required",
+    )
+
+    def migrationsToString(
+        migrations: java.util.List[org.flywaydb.core.api.output.RepairOutput]
+    ): String = {
+      if (migrations.isEmpty) {
+        "<none>"
+      } else {
+        migrations.asScala.map(_.version).mkString(", ")
+      }
+    }
+
+    // Expected output is:
+    // - All migrations below the baseline are DELETED
+    // - The migration with exactly the baseline version is ALIGNED
+    // - No migrations are REMOVED
+    val actionsTaken = flyway.repair()
```

Comment:
just the fact that we only repair if we have older migrations does not prevent us to repair accidentally new migrations as well (imagine a current prod customer updates only in 2025).
I have a hard time to reason about corner cases here, but would that not be an accidental relaxation of validation as going forward? 
Can we constrain the repairing effect only to the older migrations?